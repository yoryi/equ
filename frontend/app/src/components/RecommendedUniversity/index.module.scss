@import "src/styles/mixins/breakpoints";

:local(.card) {
  padding: 16px;

  @include breakpoint('small') {
    padding: 4px;
  }

  :local(.container) {
    display: flex;
    flex-direction: column;
    align-items: center;

    :local(.badgeContainer) {
      width: 100%;
    }

    :local(.profilePhoto) {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 64px;
      height: 64px;
      border-radius: 50%;
      box-shadow: 0 2px 4px rgba(31, 36, 43, 0.12), 0 0 8px rgba(31, 36, 43, 0.08);

      span {
        font-family: Telegraf, sans-serif;
        font-size: 24px;
        line-height: 24px;
        font-weight: bold;
        color: black;
      }
    }

    :local(.details) {
      flex: 1;
      display: flex;
      flex-direction: column;
      align-items: center;

      :local(.name) {
        margin-top: 12px;
        margin-bottom: -4px;
        font-size: 24px;
        line-height: 32px;
        font-weight: bold !important;
        color: var(--main-dark);
        letter-spacing: -0.04em;
        text-align: center;

        &:hover {
          text-decoration: underline;
        }

        @include breakpoint('small') {
          font-size: 20px;
          line-height: 22px;
          margin-bottom: 0;
        }
      }

      span {
        text-align: center;

        @include breakpoint('small') {
          font-size: 12px;
          line-height: 18px;
        }
      }
    }

    :local(.button) {
      margin-top: 32px;

      @include breakpoint('small') {
        min-width: auto !important;
        padding: 8px 34px;
        margin-bottom: 28px;
      }
    }
  }
}
