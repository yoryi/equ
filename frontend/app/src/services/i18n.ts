import filesize from "filesize"
import i18n from "i18next"
import _ from "lodash"
import { initReactI18next } from "react-i18next"

import { en } from "../translations"

void i18n.use(initReactI18next).init({
  ns: [`common`, `screens`],
  defaultNS: `screens`,
  resources: {
    en,
  },
  lng: `en`,
  fallbackLng: `en`,
  interpolation: {
    format: (value, format, lng) => {
      if (!format) {
        return value
      }

      return format.split(`,`).reduce((value: string, format: string) => {
        switch (format.trim()) {
          case `localeString`:
            return value.toLocaleString()
          case `enIndefiniteArticle`:
          case `enIndefiniteArticleCapitalized`: {
            if (!lng || lng !== `en`) {
              return value
            }

            const letters = [`a`, `e`, `i`, `o`, `u`, `h`]
            const firstLetter = value.substring(0, 1)
            let article = ``
            if (
              letters.find((letter) => firstLetter.toLowerCase() === letter)
            ) {
              article =
                format === `enIndefiniteArticleCapitalized` ? `An` : `an`
            } else {
              article = format === `enIndefiniteArticleCapitalized` ? `A` : `a`
            }

            return `${article} ${value}`
          }
          case `title`: {
            return value
              .toLowerCase()
              .split(` `)
              .map((word: string) => _.upperFirst(word))
              .join(` `)
          }
          case `lowerCase`: {
            return value.toLowerCase()
          }
          case `fileSize`: {
            return filesize(parseInt(value), { round: 0 })
          }
          default:
            return value
        }
      }, value)
    },
  },
})

export default i18n
