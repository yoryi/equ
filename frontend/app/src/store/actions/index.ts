export * from "./admin"
export * from "./attachments"
export * from "./auth"
export * from "./follow"
export * from "./help"
export * from "./journeys"
export * from "./media"
export * from "./notifications"
export * from "./profile"
export * from "./searchAndFilters"
export * from "./settings"
export * from "./shared"
export * from "./student"
export * from "./surveys"
export * from "./university"
