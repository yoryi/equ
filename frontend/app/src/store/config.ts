interface Config {
  baseURL: string | null
}

export const config: Config = {
  baseURL: null,
}

export const initReduxStore = (baseURL: string) => {
  config.baseURL = baseURL
}
