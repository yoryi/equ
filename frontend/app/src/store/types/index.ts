export * from "./admin"
export * from "./auth"
export * from "./errors"
export * from "./ethnicity"
export * from "./follow"
export * from "./gender"
export * from "./hardship"
export * from "./help"
export * from "./highSchool"
export * from "./interest"
export * from "./notifications"
export * from "./profile"
export * from "./race"
export * from "./search"
export * from "./settings"
export * from "./shared"
export * from "./store"
export * from "./student"
export * from "./survey"
export * from "./university"
export * from "./utils"
