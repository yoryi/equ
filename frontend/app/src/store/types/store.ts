import { AdminReducer } from "../reducers/admin/reducer"
import { AuthReducer } from "../reducers/auth/reducer"
import { FollowReducer } from "../reducers/follow"
import { MediaReducer } from "../reducers/media/reducer"
import { NotificationsReducer } from "../reducers/notifications"
import { ProfileReducer } from "../reducers/profile/reducer"
import { SearchAndFiltersReducer } from "../reducers/searchAndFilters/reducer"
import { SettingsReducer } from "../reducers/settings/reducer"
import { SharedReducer } from "../reducers/shared/reducer"
import { StudentReducer } from "../reducers/student"
import { SurveysReducer } from "../reducers/surveys/reducer"
import { UniversityReducer } from "../reducers/university"

export interface ReduxState {
  auth: AuthReducer
  media: MediaReducer
  notifications: NotificationsReducer
  profile: ProfileReducer
  shared: SharedReducer
  settings: SettingsReducer
  student: StudentReducer
  surveys: SurveysReducer
  university: UniversityReducer
  follow: FollowReducer
  searchAndFilters: SearchAndFiltersReducer
  admin: AdminReducer
}
