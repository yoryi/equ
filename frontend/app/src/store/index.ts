import * as actions from "./actions"
import { initReduxStore } from "./config"
import { store } from "./createStore"

export { actions,initReduxStore, store }
