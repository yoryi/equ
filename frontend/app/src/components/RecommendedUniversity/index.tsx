import * as mui from "@material-ui/core"
import classNames from "classnames"
import { MDBCard, MDBCardBody } from "mdbreact"
import * as React from "react"
import { useTranslation } from "react-i18next"
import { Link } from "react-router-dom"

import { MATCH_LEVELS } from "../../const/matchLevels"
import { UniversityProfile } from "../../store/types"
import { formatMatchingScoreLabel } from "../../utils/matchingScore"
import { nameAbbreviation } from "../../utils/nameAbbreviation"
import Badge from "../Badge"
import Button from "../Button/Button"
import Styles from "./index.module.scss"

interface RecommendedUniversityBaseProps {
  className?: string
}

interface RecommendedUniversityProps {
  university: UniversityProfile
  onClick?: () => void
  loading?: undefined
}

interface LoadingRecommendedUniversityProps {
  university?: undefined
  onClick?: undefined
  loading: true
}

const RecommendedUniversity: React.FC<
  RecommendedUniversityBaseProps &
    (RecommendedUniversityProps | LoadingRecommendedUniversityProps)
> = ({ className, university, onClick, loading }) => {
  const { t } = useTranslation(`common`)

  const matchLevel =
    university &&
    MATCH_LEVELS.find(
      ({ min, max }) => university.score >= min && university.score < max,
    )

  return (
    <MDBCard className={classNames(Styles.card, className)}>
      <MDBCardBody className={Styles.container}>
        {!!matchLevel && !!university && (
          <div className={Styles.badgeContainer}>
            <Badge variant={matchLevel.color}>
              {formatMatchingScoreLabel(university.score, t)}
            </Badge>
          </div>
        )}

        {!loading && (
          <div className={Styles.profilePhoto}>
            <span>{nameAbbreviation(university?.name ?? ``, true)}</span>
          </div>
        )}
        {loading && <mui.Skeleton width={64} height={64} variant="circular" />}

        <div className={Styles.details}>
          {!loading && (
            <>
              <Link
                className={Styles.name}
                to={`/university/${university?.id}`}
              >
                {university?.name}
              </Link>
              <span
                className={`text-3`}
              >{`${university?.city}, ${university?.state}`}</span>
            </>
          )}
          {loading && (
            <>
              <h3 className={`mb-2 mb-md-0`}>
                <mui.Skeleton width={150} />
              </h3>
              <span>
                <mui.Skeleton width={200} />
              </span>
            </>
          )}
        </div>

        {!loading && (
          <Button className={Styles.button} size={`xs`} onClick={onClick}>
            {t(`follow`)}
          </Button>
        )}
        {loading && (
          <mui.Skeleton className={Styles.button} width={125} height={40} />
        )}
      </MDBCardBody>
    </MDBCard>
  )
}

export default RecommendedUniversity
