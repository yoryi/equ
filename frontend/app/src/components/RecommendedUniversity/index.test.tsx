//Testing
import { act, fireEvent,render } from "@testing-library/react"
import faker from "faker"
import * as React from "react"
//Translations
import { I18nextProvider } from "react-i18next"
//Navigation
import { MemoryRouter } from "react-router"

import i18n from "../../services/i18n"
//Types
import { UniversityProfile } from "../../store/types"
//Components
import RecommendedUniversity from "./"

describe(`RecommendedUniversity component`, () => {
  faker.seed(1000)

  const wrapper: React.FC = ({ children }) => (
    <MemoryRouter>
      <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
    </MemoryRouter>
  )

  let university: UniversityProfile
  beforeEach(() => {
    university = {
      id: faker.random.number(10000),
      name: `${faker.lorem.word()} University`,
      city: faker.address.city(),
      state: faker.address.stateAbbr(),
      avatar: faker.image.imageUrl(),
      score: faker.random.number({ min: 75, max: 100 }),
      email: faker.internet.email(),
      website: faker.internet.url(),
    }
  })

  describe(`should render and match the snapshot`, () => {
    it(`university`, () => {
      const { container } = render(
        <RecommendedUniversity university={university} />,
        { wrapper },
      )

      expect(container.firstChild).toMatchSnapshot()
    })

    it(`loading`, () => {
      const { container } = render(<RecommendedUniversity loading />, {
        wrapper,
      })

      expect(container.firstChild).toMatchSnapshot()
    })
  })

  it(`should handle following the university`, () => {
    const onClick = jest.fn()

    const { getByText } = render(
      <RecommendedUniversity university={university} onClick={onClick} />,
      { wrapper },
    )

    act(() => {
      fireEvent.click(getByText(i18n.t(`common:follow`)))
    })

    expect(onClick).toHaveBeenCalled()
  })
})
