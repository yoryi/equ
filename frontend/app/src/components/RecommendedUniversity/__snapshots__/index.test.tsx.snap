// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommendedUniversity component should render and match the snapshot loading 1`] = `
<div
  class="card card"
  data-test="card"
>
  <div
    class="card-body container"
    data-test="card-body"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-circular MuiSkeleton-pulse css-fomgco-MuiSkeleton-root"
      style="width: 64px; height: 64px;"
    />
    <div
      class="details"
    >
      <h3
        class="mb-2 mb-md-0"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          style="width: 150px;"
        />
      </h3>
      <span>
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          style="width: 200px;"
        />
      </span>
    </div>
    <span
      class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse button css-1n7gpw1-MuiSkeleton-root"
      style="width: 125px; height: 40px;"
    />
  </div>
</div>
`;

exports[`RecommendedUniversity component should render and match the snapshot university 1`] = `
<div
  class="card card"
  data-test="card"
>
  <div
    class="card-body container"
    data-test="card-body"
  >
    <div
      class="badgeContainer"
    >
      <div
        class="container red"
      >
        Average Match (84%)
      </div>
    </div>
    <div
      class="profilePhoto"
    >
      <span>
        QU
      </span>
    </div>
    <div
      class="details"
    >
      <a
        class="name"
        href="/university/6536"
      >
        quia University
      </a>
      <span
        class="text-3"
      >
        New Valentinatown, MO
      </span>
    </div>
    <button
      class="primary-button button default-button xs-button blue-button"
    >
      Follow
    </button>
  </div>
</div>
`;
